src/notification.rs

```diff
@@ -4,75 +4,54 @@ use data::{
     user::{Nick, NickRef},
 };
 
-use crate::audio;
-
 pub use self::toast::prepare;
 
 mod toast;
 
-pub fn connected(
-    config: &config::Notifications<Sound>,
-    audio: &mut audio::State,
-    server: impl ToString,
-) {
-    show_notification(&config.connected, audio, "Connected", server);
+pub fn connected(config: &config::Notifications<Sound>, server: impl ToString) {
+    show_notification(&config.connected, "Connected", server);
 }
 
-pub fn reconnected(
-    config: &config::Notifications<Sound>,
-    audio: &mut audio::State,
-    server: impl ToString,
-) {
-    show_notification(&config.reconnected, audio, "Reconnected", server);
+pub fn reconnected(config: &config::Notifications<Sound>, server: impl ToString) {
+    show_notification(&config.reconnected, "Reconnected", server);
 }
 
-pub fn disconnected(
-    config: &config::Notifications<Sound>,
-    audio: &mut audio::State,
-    server: impl ToString,
-) {
-    show_notification(&config.disconnected, audio, "Disconnected", server);
+pub fn disconnected(config: &config::Notifications<Sound>, server: impl ToString) {
+    show_notification(&config.disconnected, "Disconnected", server);
 }
 
-pub fn highlight(
-    config: &config::Notifications<Sound>,
-    audio: &mut audio::State,
-    nick: NickRef,
-    channel: String,
-) {
+pub fn highlight(config: &config::Notifications<Sound>, nick: NickRef, channel: String) {
     show_notification(
         &config.highlight,
-        audio,
         "Highlight",
         format!("{} highlighted you in {}", nick, channel),
     );
 }
 
 pub fn file_transfer_request(
     config: &config::Notifications<Sound>,
-    audio: &mut audio::State,
     nick: Nick,
     server: impl ToString,
 ) {
     show_notification(
         &config.file_transfer_request,
-        audio,
         &format!("File transfer from {}", nick),
         server,
     );
 }
 
-fn show_notification(
-    notification: &notification::Loaded,
-    audio: &mut audio::State,
-    title: &str,
-    body: impl ToString,
-) {
+fn show_notification(notification: &notification::Loaded, title: &str, body: impl ToString) {
     if notification.show_toast {
         toast::show(title, body);
     }
 
     if let Some(sound) = &notification.sound {
-        audio.play(sound);
+        let sound = sound.clone();
+
+        tokio::spawn(async move {
+            if let Err(error) = sound.play() {
+                log::debug!("Unable to play notification sound: {error}");
+            }
+        });
```

Comment:
We should not be blocking in an async thread. We either need to just use thread::spawn or use spawn_blocking inside the spawned future.

Reply:
I've fixed in ada5870 and moved rodio dep to frontend so data isn't dependent on it

Reply:
Ahh, thank you for catching this!  I initially wrote it as `thread::spawn` but changed it to `tokio::spawn` since I wasn't sure if it was ok to mix `std::thread` with the `tokio` stuff.  And... I neglected to check whether they were actually equivalent.  Sorry about that!

Reply:
I don't think it'll be an issue, these are real threads so the OS is responsible over scheduling and if we have all cores blocked via tokio worker cores + this thread, the OS can handle that (I'm pretty sure)